library/src/containers/AsyncApi/AsyncApi.tsx

```diff
@@ -1,176 +1,171 @@
-            <TopicsComponent
-              baseTopic={validatedSchema.baseTopic}
-              topics={validatedSchema.topics}
-            />,
+          {concatenatedConfig.show.security &&
+            validatedSchema.components &&
+            validatedSchema.components.securitySchemes && (
+              <SecurityComponent
+                security={Object.keys(
+                  validatedSchema.components.securitySchemes,
+                ).map(
+                  elem =>
+                    validatedSchema.components &&
+                    validatedSchema.components.securitySchemes &&
+                    validatedSchema.components.securitySchemes[elem],
+                )}
+              />
+            )}
+
+          {concatenatedConfig.show.channels && !!validatedSchema.channels && (
+            <Channels channels={validatedSchema.channels} />
           )}
           {validatedSchema.components && (
             <>
-              {this.showComponent(
-                concatenatedConfig.show.messages &&
-                  Boolean(validatedSchema.components) &&
-                  Boolean(validatedSchema.components!.messages),
-                <MessagesComponent
-                  messages={validatedSchema.components!.messages}
-                />,
-              )}
-              {this.showComponent(
-                concatenatedConfig.show.schemas &&
-                  Boolean(validatedSchema.components) &&
-                  Boolean(validatedSchema.components!.schemas),
-                <SchemasComponent
-                  schemas={validatedSchema.components!.schemas}
-                />,
-              )}
+              {concatenatedConfig.show.messages &&
+                Boolean(validatedSchema.components) &&
+                Boolean(validatedSchema.components.messages) && (
+                  <MessagesComponent
+                    messages={validatedSchema.components.messages}
+                  />
+                )}
+              {concatenatedConfig.show.schemas &&
+                Boolean(validatedSchema.components) &&
+                Boolean(validatedSchema.components.schemas) && (
+                  <SchemasComponent
+                    schemas={validatedSchema.components.schemas}
+                  />
+                )}
             </>
           )}
         </AsyncApiWrapper>
       </ThemeProvider>
     );
   }
+
+  private async parseSchema(schema: PropsSchema) {
+    if (isFetchingSchemaInterface(schema)) {
+      const { data, error } = await parser.parseFromUrl(schema);
+
+      const beautifiedSchema = this.beautifySchema(data);
+
+      this.setState({ validatedSchema: beautifiedSchema, error });
+    } else {
+      const { data, error } = await parser.parse(schema);
+
+      const beautifiedSchema = this.beautifySchema(data);
+
+      this.setState({ validatedSchema: beautifiedSchema, error });
+    }
+  }
+
+  private beautifySchema(schema: NullableAsyncApi): NullableAsyncApi {
+    if (!schema) {
```

Comment:
I'd rather have "dirty" types here than in beautifier, because if I move it there then beautifier will have to have option to return `null`, which in turn enables each beautifier to be rewritten to return `null`, which adds logic that this simple facade here